samtranslator/model/eventsources/push.py

```diff
@@ -218,6 +218,8 @@ class S3(PushEventSource):
     def resources_to_link(self, resources):
         if isinstance(self.Bucket, dict) and "Ref" in self.Bucket:
             bucket_id = self.Bucket["Ref"]
+            if not isinstance(bucket_id, str):
```

Comment:
As discussed offline, we should use `six` for string types to deal with Python 2's `unicode` et al.